### DISCUSSION TITLE
@soroban-react/utils, contracts, wallet-data, connect-button, utils : packages for ReactJS dApps!

### DISCUSSION THREAD

**Discussion by esteblock** [stellar/sorobanathon#163] [General]
Hello, I continued with the support of the  [@soroban-react](https://github.com/esteblock/soroban-react/) library.
Now added the following packages:

|Package|Version|
|--|--|
|[`@soroban-react/contracts`](https://github.com/esteblock/soroban-react/tree/main/packages/contracts)                | [![npm version](https://img.shields.io/npm/v/@soroban-react/contracts/latest.svg)](https://www.npmjs.com/package/@soroban-react/contracts/v/latest)|
|[`@soroban-react/utils`](https://github.com/esteblock/soroban-react/tree/main/packages/utils)                     | [![npm version](https://img.shields.io/npm/v/@soroban-react/utils/latest.svg)](https://www.npmjs.com/package/@soroban-react/utils/v/latest)|
|[`@soroban-react/wallet-data`](https://github.com/esteblock/soroban-react/tree/main/packages/core)                     | [![npm version](https://img.shields.io/npm/v/@soroban-react/wallet-data/latest.svg)](https://www.npmjs.com/package/@soroban-react/wallet-data/v/latest)|
|[`@soroban-react/connect-button`](https://github.com/esteblock/soroban-react/tree/main/packages/core)                     | [![npm version](https://img.shields.io/npm/v/@soroban-react/connect-button/latest.svg)](https://www.npmjs.com/package/@soroban-react/connect-button/v/latest)|

Each package do different things, but in my opinion the most important here are the @soroban-react/contracts and  @soroban-react/wallet-data. The first one provides all the funtions to interact with smart contracts (read and write), using the web browser/wallet extension. The second one is provides the typical "Connect Wallet" button, and when connected, it does show the information about the wallet.

This packages are currently being used in the [*pet-adopt-soroban*](https://github.com/esteblock/pet-adopt-soroban) project
I am opening issues in [stellar/soroban-react-example](https://github.com/stellar/soroban-example-dapp/) repo in order to include them there as well (see https://github.com/stellar/soroban-example-dapp/issues/72 ) 

To do this, I took code from the  [stellar/soroban-react-example](https://github.com/stellar/soroban-example-dapp/) repo (and did some coding as well, in order to transform it into a library)

Now, front-end devs don't need to copy-paste all the time.
And it will be easier to deal with new versions of sdk, etc...

I hope you like!
Please add issues!

**Comment by kalepail** [comment]
Haven't dug into it all (yet) but this seems like an impressive amount of work! Great job! 👏

**Reply by esteblock** [reply]
Thanks! Now created PR in soroban-example-dapp to use @soroban-react/contracts library

https://github.com/stellar/soroban-example-dapp/pull/79

**Comment by achallagundla** [comment]
Heya,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 2000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Reply by esteblock** [reply]
> Heya, Congratulations! Your contribution to Sorobanathon: First Light has been accepted! tada Your submission may qualify for a reward of 2000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.
> 
> Please let us know if you have any questions. (You can also message me on discord @ anuxhya)
> 
> All the best, Anuhya
> 
> PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

Thanks! Sent you a DM in Discord, @achallagundla.
Waiting for that answer before submiting the form. Thanks!
